lock-api/src/main/java/com/palantir/lock/client/TimeLockClient.java

```diff
@@ -88,16 +91,19 @@ public TimestampRange getFreshTimestamps(int numTimestampsRequested) {
     @Override
     public LockImmutableTimestampResponse lockImmutableTimestamp() {
         LockImmutableTimestampResponse response = executeOnTimeLock(delegate::lockImmutableTimestamp);
-        lockRefresher.registerLock(response.getLock());
+        lockRefresher.registerLocks(ImmutableList.of(response.getLock()));
         return response;
     }
 
     @Override
-    public StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction() {
-        StartIdentifiedAtlasDbTransactionResponse response = executeOnTimeLock(
-                delegate::startIdentifiedAtlasDbTransaction);
-        lockRefresher.registerLock(response.immutableTimestamp().getLock());
-        return response;
+    public List<StartIdentifiedAtlasDbTransactionResponse> startIdentifiedAtlasDbTransactionBatch(int count) {
+        List<StartIdentifiedAtlasDbTransactionResponse> responses = executeOnTimeLock(
+                () -> delegate.startIdentifiedAtlasDbTransactionBatch(count));
+        lockRefresher.registerLocks(responses
```

Comment:
You're relying on this not throwing. I think it's fine in this refactoring, but need to track fixing this, same as the other bug.